test/test-fs-readdir.c

```diff
@@ -0,0 +1,514 @@
+/* Copyright Joyent, Inc. and other Node contributors. All rights reserved.
```

Comment:
I don't mind doing this but all other tests have the copyright currently being used in the PR.